convert_hf_to_gguf.py

```diff
@@ -1570,6 +1570,34 @@ def modify_tensors(self, data_torch: Tensor, name: str, bid: int | None) -> Iter
         return [(self.map_tensor_name(name), data_torch)]
 
     def prepare_tensors(self):
+        if rope_scaling := self.find_hparam(["rope_scaling"], optional=True):
+            if rope_scaling.get("rope_type", '').lower() == "llama3":
+                base = self.hparams.get("rope_theta", 10000.0)
+                dim = int((self.hparams["hidden_size"] // self.hparams["num_attention_heads"]) * self.hparams.get("partial_rotary_embeddings", 1.0))
```

Comment:
Where does `partial_rotary_embeddings` come from? I don't see it in Llama-3.1's `config.json`, and if present, it would mess with the rope_freqs tensor size in a way which is not handled when loading it from `src/llama.cpp`.

```suggestion
                dim = self.hparams["hidden_size"] // self.hparams["num_attention_heads"]
```

Reply:
Good catch! Sorry!